### PR TITLE
nrf52_bsim: Use the flash simulator by default

### PR DESCRIPTION
**Pull request by aescolar**
Today there is no "FLASH" peripheral in the nrf52_bsim, so the real nrf flash driver cannot be used.
But the flash simulator can be used with the nrf52_bsim just like with any other board (it does not even require the POSIX arch, even though, for this arch, it is even nicer, as it allows you to keep the flash content in a file in disk).

Today users need to add a DT overlay to their app if they want to use the flash simulator with this board, which is a bit cumbersome.

This PR provides a good default dts configuration for most use cases for this board, which makes it much easier for users to enable the flash simulator in their apps/tests.

------
    nrf52_bsim: dts: Enable the flash simulator
    
    So when applications try to use CONFIG_FLASH it will
    work and default to the simulator without the need for
    any overlay.
------
    tests/net: For nrf52_bsim use FLASH simulator in RAM
    
    Instead of falling back to the openthread RAM settings backend
Note that for these tests we keep the flash just in RAM because we do not want to persist anything between runs in this case.

### CODE REVIEW

File: drivers/flash/flash_simulator.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (de-nordic) [inline drivers/flash/flash_simulator.c:2]: we usually extend copyrights, i.e.: `2018-2023`

**Comment by aescolar** [comment]
CC @canisLupus1313

**Review by canisLupus1313 — DISMISSED** [review]

File: boards/posix/nrf52_bsim/nrf52_bsim.dts [hunk 2]
```diff
@@ -64,6 +59,46 @@
...
+			reg = <0x00000000 DT_SIZE_K(2048)>;
+
+			partitions {
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				storage_partition: partition@00000 {
+					label = "storage";
+					reg = <0x0000000 0x00004000>;
```
Comment (de-nordic) [inline boards/posix/nrf52_bsim/nrf52_bsim.dts:91]: You could probably use the DT_SIZE_K macro too. Seems that you define quite big flash (2 meg?) and use 16k only.

**Review by de-nordic — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Comment by aescolar** [comment]
Closing as superseded by https://github.com/zephyrproject-rtos/zephyr/pull/57346

**Pull request closed without merging**